src/core/Darwin.zig

```diff
@@ -86,23 +94,62 @@ pub fn init(
         surface_descriptor.next_in_chain = .{ .from_metal_layer = metal_descriptor };
 
         const screen = objc.app_kit.Screen.mainScreen();
-        const rect = objc.core_graphics.Rect{ // TODO: use a meaningful rect
+
+        var rect = objc.core_graphics.Rect{
             .origin = .{ .x = 100, .y = 100 },
-            .size = .{ .width = 480, .height = 270 },
+            .size = .{ .width = @floatFromInt(options.size.width), .height = @floatFromInt(options.size.height) },
         };
+
+        if (screen) |s| {
+            const frame = s.visibleFrame();
+            rect.origin.x = frame.size.width / 2.0 - @as(@TypeOf(rect.origin.x), @floatFromInt(options.size.width)) / 2.0;
+            rect.origin.y = frame.size.height / 2.0 - @as(@TypeOf(rect.origin.y), @floatFromInt(options.size.height)) / 2.0;
+        }
+
         const window_style =
             (if (options.display_mode == .fullscreen) objc.app_kit.WindowStyleMaskFullScreen else 0) |
             (if (options.display_mode == .windowed) objc.app_kit.WindowStyleMaskTitled else 0) |
             (if (options.display_mode == .windowed) objc.app_kit.WindowStyleMaskClosable else 0) |
             (if (options.display_mode == .windowed) objc.app_kit.WindowStyleMaskMiniaturizable else 0) |
             (if (options.display_mode == .windowed) objc.app_kit.WindowStyleMaskResizable else 0);
-        window = objc.app_kit.Window.alloc().initWithContentRect_styleMask_backing_defer_screen(rect, window_style, objc.app_kit.BackingStoreBuffered, true, screen);
-        window.?.setReleasedWhenClosed(false);
-        if (window.?.contentView()) |view| {
-            view.setLayer(@ptrCast(layer));
+
+        window_opt = objc.app_kit.Window.alloc().initWithContentRect_styleMask_backing_defer_screen(
+            rect,
+            window_style,
+            objc.app_kit.BackingStoreBuffered,
+            true,
+            screen,
+        );
+        if (window_opt) |window| {
+            window.setReleasedWhenClosed(false);
+            if (window.contentView()) |view| {
+                view.setLayer(@ptrCast(layer));
+            }
+            window.setIsVisible(true);
+            window.makeKeyAndOrderFront(null);
+
+            const delegate = objc.mach.WindowDelegate.allocInit();
+
+            const Helper = struct {
+                pub fn cCallback_windowWillResize_toSize(block: *objc.foundation.BlockLiteral(*Darwin), size: objc.app_kit.Size) callconv(.C) void {
+                    const self: *Darwin = block.context;
```

Comment:
Gotcha!

Also on this same line, would it be preferred since we are going to have a decent amount of callbacks to just move `Helper` down to the bottom of the file and name it `WindowDelegateCallbacks` or something, so its not implementing all the logic of those callbacks in the middle of `init`?

Reply:
I went ahead and moved it, I think it will be a better organization to have these callbacks outside of `init`, will change if you prefer.